rclrs/src/lib.rs

```diff
@@ -25,95 +25,4 @@ pub use self::context::*;
 pub use self::error::*;
 pub use self::node::*;
 pub use self::qos::*;
-
-use self::rcl_bindings::*;
-
-use wait::WaitSet;
-
-/// Wrapper around [`spin_once`]
-pub fn spin(node: &node::Node) -> Result<(), RclReturnCode> {
-    while unsafe { rcl_context_is_valid(&mut *node.context.lock() as *mut _) } {
-        if let Some(error) = spin_once(node, 500).err() {
-            match error {
-                RclReturnCode::Timeout => continue,
-                error => return Err(error),
-            };
-        }
-    }
-
-    Ok(())
-}
-
-/// Main function for waiting.
-///
-/// Following is a schematic representation of the interation of [`spin_once`] with ROS RCL FFI
-///
-/// +---------------------------------------+
-/// |                                       |
-/// |   rcl_get_zero_initialized_wait_set   |
-/// |                                       |
-/// +-----------------+---------------------+
-///                   |
-///                   |
-///        +----------v----------+
-///        |  rcl_wait_set_init  |
-///        +----------+----------+
-///                   |
-///        +----------v----------+
-///        |  rcl_wait_set_clear |
-///        +----------+----------+
-///                   |
-///   +---------------v------------------+   <---+
-///   |rcl_wait_set_add_{subscription,   |       | for all subscriptions,
-///   |                  guard_condition |       | services, etc.
-///   |                  timer,          |       |
-///   |                  client,         |       |
-///   |                  service,        |       |
-///   |                  e^ent}          |       |
-///   +---------------+------------------+ +-----+
-///                   |
-///         +---------v----------+
-///         |     rcl_wait       |
-///         +---------+----------+
-///                   |
-///         +---------v----------+
-///         | rcl_wait_set_fini  |
-///         +--------------------+
-///
-///
```

Comment:
I rather liked the ASCII art representation of how the underlying RCL functions were supposed to work/be called. Would it be worth adding this back in to where `spin_once` now lives?

Reply:
To be honest, I don't find it that helpful:
- All of the functions have different names in `rclrs` than in `rcl`
- `rcl_get_zero_initialized_wait_set` and `rcl_wait_set_fini` are implementation details and should not concern users
- `rcl_wait_set_clear` doesn't need to be called after `init`. If anything, it should be an arrow back from a filled wait set to an empty one.

So basically, it just says "add subscriptions before you call `wait()`", and that is mentioned in a comment already. 

If we want something like this for contributors to learn about the typical `get zero-initialized value -> init -> do something -> fini` sequence, that's a fair point, but I think I'd add it to `CONTRIBUTING.md` since it's not specific to the wait set.